src/ballistics.cpp

```diff
@@ -231,9 +232,18 @@ dealt_projectile_attack projectile_attack( const projectile &proj_arg,
     creature_tracker &creatures = get_creature_tracker();
     Creature *target_critter = creatures.creature_at( target_arg );
     map &here = get_map();
-    double target_size = target_critter != nullptr ?
-                         target_critter->ranged_target_size() :
-                         here.ranged_target_size( target_arg );
+    double target_size;
+    if( target_critter != nullptr ) {
+        const monster *mon = target_critter->as_monster();
+        if( !mon || !proj_arg.proj_effects.count( ammo_effect_WIDE ) ) {
+            target_size = target_critter->ranged_target_size();
+        } else {
+            // ammo with ammo_effect_WIDE ignores mon_flag_HARDTOSHOOT
+            target_size = occupied_tile_fraction( mon->get_size() );
+        }
```

Comment:
Readability: The condition checked in the first branch of the `if` is only semantically relevant to the second, which makes the clause mildly confusing. Consider either dropping the `else`:

```suggestion
        target_size = target_critter->ranged_target_size();
        if( mon && proj_arg.proj_effects.count( ammo_effect_WIDE ) ) {
            // ammo with ammo_effect_WIDE ignores mon_flag_HARDTOSHOOT
            target_size = occupied_tile_fraction( mon->get_size() );
        }
```

or at least reordering the branches:

```cpp
        if( mon && proj_arg.proj_effects.count( ammo_effect_WIDE ) ) {
            // ammo with ammo_effect_WIDE ignores mon_flag_HARDTOSHOOT
            target_size = occupied_tile_fraction( mon->get_size() );
        } else {
            target_size = target_critter->ranged_target_size();
        }
```

Reply:
Well, I was thinking that worte that way can make it faster when !mon, but you are right, that would be confusing.